src/main/scala/no/ndla/imageapi/service/WriteService.scala

```diff
@@ -60,14 +60,24 @@ trait WriteService {
       }
     }
 
+    def using[R <: Closeable, T](stream: R)(f: R => T): T =
+      try {
+        f(stream)
+      } finally {
+        stream.close()
+      }
+
     private[service] def uploadImage(file: FileItem): Try[Image] = {
       val extension = getFileExtension(file.name).getOrElse("")
       val contentType = file.getContentType.getOrElse("")
       val fileName = Stream.continually(randomFileName(extension)).dropWhile(imageStorage.objectExists).head
 
-      imageStorage.uploadFromStream(new ByteArrayInputStream(file.get), fileName, contentType, file.size).map(filePath => {
-        Image(filePath, file.size, contentType)
-      })
+      using(new ByteArrayInputStream(file.get)) {
+        stream =>
+          imageStorage.uploadFromStream(stream, fileName, contentType, file.size).map(filePath => {
+            Image(filePath, file.size, contentType)
+          })
+      }
```

Comment:
This approach seems plausible, but is it possible to do this logic in the `uploadFromStream`-method. See the method `AmazonImageStorageService.uploadFromUrl` which also uses a stream.

Reply:
hmm...but that stream will probably be closed from the client..